api/src/user/user.controller.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpException,
+  Param,
+  ParseIntPipe,
+  Post,
+  Put,
+  Query,
+} from '@nestjs/common';
+import { Prisma } from '@prisma/client';
+import { UserService } from './user.service';
+
+@Controller('user')
+export class UserController {
+  constructor(private readonly userService: UserService) {}
+
+  @Get()
+  async getUsers(
+    @Query('filter') filter: string,
+    @Query('sort') sort: string,
+    @Query('range') range: string,
+  ) {
+    let params;
+    const filters = JSON.parse(filter);
+    if (filters.email) {
+      params = {
+        where: {
+          email: {
+            contains: filters.email,
+          },
+        },
+      };
+    }
+
+    const rangeNumber = JSON.parse(range) as number[];
+    if (rangeNumber) {
+      params = {
+        ...params,
+        take: rangeNumber[1] - rangeNumber[0],
+        skip: rangeNumber[0],
+      };
+    }
+
+    const sortParsed = JSON.parse(sort);
+    if (sortParsed) {
+      params = {
+        ...params,
+        orderBy: {
+          [sortParsed[0]]: sortParsed[1].toLowerCase(),
+        }
+      };
+    }
+    
+    const users = await this.userService.getUsers(params);
+    return users;
+  }
+
+  @Get(':id')
+  async getUser(@Param('id', ParseIntPipe) id: number) {
+    if (!id) {
```

Comment:
how can there be no id? If the route was matched, there should be one, right?

Reply:
You are right, and the pipe prevents from passing anything, because it raises an error if you do not get an number